pkg/controllers/dynakube/proxy/reconciler_test.go

```diff
@@ -241,53 +272,23 @@ func TestReconcileProxyValueFrom(t *testing.T) {
 	var proxyUrl = buildProxyUrl(proxyHttpScheme, proxyUsername, proxyPassword, proxyHost, proxyPort)
 
 	var testClient = fake.NewClientBuilder().WithObjects(createProxySecret(proxyUrl)).Build()
-	r := newTestReconcilerWithInstance(testClient)
 
 	t.Run(`reconcile proxy ValueFrom`, func(t *testing.T) {
-		r.dk.Spec.Proxy = &dynakube.DynaKubeProxy{ValueFrom: customProxySecret}
-		err := r.Reconcile(context.Background())
-		require.NoError(t, err)
-
-		var proxySecret corev1.Secret
-		_ = r.client.Get(context.Background(), client.ObjectKey{Name: BuildSecretName(testDynakubeName), Namespace: testNamespace}, &proxySecret)
-
-		assert.Equal(t, []byte(proxyPassword), proxySecret.Data[passwordField])
-		assert.Equal(t, []byte(proxyPort), proxySecret.Data[portField])
-		assert.Equal(t, []byte(proxyHost), proxySecret.Data[hostField])
-		assert.Equal(t, []byte(proxyUsername), proxySecret.Data[usernameField])
-		assert.Equal(t, []byte(proxyHttpScheme), proxySecret.Data[schemeField])
-	})
-	t.Run(`Change of Proxy ValueFrom to Value`, func(t *testing.T) {
```

Comment:
Whats the reason for merging these two unit tests?

Reply:
I removed this test because I do not think it is properly doable as we mock the reconciler here. This means we do not have access to the incorporated dynakube within the reconciler and change the Proxy of it, at least I did not find a way to do this properly

Reply:
Update: I found a way to do it properly thanks to @0sewa0 

I reintroduced the test.

done: https://github.com/Dynatrace/dynatrace-operator/pull/3875/commits/1c4abfe681211d132ec48698079d2618271bdec9

Reply:
Niiice thanks!